### ISSUE TITLE
Suggestion: drag tile to place object (on object layer)

### ISSUE THREAD

**Issue by caiolopez** [bjorn/tiled#1070]
When working in an object layer, it would be great to just drag the tile from the tileset browser directly to the map, bypassing the place tool selection.

**Comment by bjorn** [comment]
I agree that would be a nice feature, but the problem with this is that dragging on the tileset view is already taken for selecting a block of multiple tiles. Some options:

* Perform the placement drag only with some modifier. Not nice.
* Move the existing behavior to the right mouse button. But that is taken by the context menu. This menu could be moved away for now. Still, all in all this will make tile layer editing less intuitive.
* Do different things with click+drag based on whether a tile layer or object layer is selected. Is this alright, or too inconsistent?

What's your opinion? Do you have other suggestions?

**Comment by caiolopez** [comment]
The last bullet pretty much nailed it. I don't think it would be an inconsistence to have the left-drag doing different things depending on the layer type, given that there is no functionality assigned for left-drag on object layers, plus, one would hardly expect to be able to place multiple objects like a brush. Finally, both types of layers are already quite different to handle -- inserting, deleting, moving, are all mapped to different keys and work in different ways. Object layers won't even preview the object to be placed on the cursor, as opposed to the

**Comment by bjorn** [comment]
Alright, thanks for your feedback! It does sound reasonable to me.

> Object layers won't even preview the selected tile to be inserted on the cursor, as opposed to the tile layers -- in fact, that one would also be a nice suggestion!

Yeah, that's a long time missing feature that I should make sure to finally get around to now that I can afford to spend more time on Tiled again. It's covered already by issue #537.

**Comment by caiolopez** [comment]
Great to hear I helped, and thank you for your feedback as well.

**Comment by caiolopez** [comment]
Any news on "DRAG TO PLACE OBJECT"? What about the "PREVIEW SELECTED OBJECT AT CURSOR"? Thank you!

**Comment by bjorn** [comment]
> Any news on "DRAG TO PLACE OBJECT"? What about the "PREVIEW SELECTED OBJECT AT CURSOR"? Thank you!

I hope the all-caps was not meant as shouting. I realize it's almost two years later, and that it hasn't been done yet, but we have done a huge amount of other things in the meantime. Everybody has their own wishes and priorities. That said, #537 is on the Roadmap for Tiled 1.2, so it is something I'm trying to get around to as soon as possible.

Note that I am currently doing [a two-week fundraising](https://www.mapeditor.org/2018/05/16/help-fund-tiled-development.html), because currently I can't afford to keep spending two days per week on Tiled. Any help is appreciated!

**Comment by caiolopez** [comment]
Haha definitely not shouting. Thank you for the quick response.
